agent/agentbootstrap/bootstrap.go

```diff
@@ -646,60 +625,6 @@ func (b *AgentBootstrap) initBootstrapNode(
 	return node, nil
 }
 
-// initControllerCloudService creates cloud service for controller service.
-func (b *AgentBootstrap) initControllerCloudService(
-	ctx stdcontext.Context,
-	cloudSpec environscloudspec.CloudSpec,
-	provider environs.EnvironProvider,
-	st *state.State,
-) error {
-	stateParams := b.stateInitializationParams
-	controllerUUID := stateParams.ControllerConfig.ControllerUUID()
-	env, err := b.getEnviron(ctx, controllerUUID, cloudSpec, stateParams.ControllerModelConfig, provider)
-	if err != nil {
-		return errors.Annotate(err, "getting environ")
-	}
-
-	broker, ok := env.(caas.ServiceManager)
-	if !ok {
-		// this should never happen.
-		return errors.Errorf("environ %T does not implement ServiceManager interface", env)
-	}
-	svc, err := broker.GetService(ctx, k8sconstants.JujuControllerStackName, true)
-	if err != nil {
-		return errors.Trace(err)
-	}
-
-	if len(svc.Addresses) == 0 {
-		// this should never happen because we have already checked in k8s controller bootstrap stacker.
-		return errors.NotProvisionedf("k8s controller service %q address", svc.Id)
-	}
-	addrs, err := svc.Addresses.ToSpaceAddresses(st)
```

Comment:
I think it is probably an error to move this method whole method (`initControllerCloudService`) out. We probably need the service entry sooner than the bootstrap worker.

Try restoring this method. Spaces are currently not supported on k8s, so instead of calling `ToSpaceAddresses`, we can create them directly with the `alpha` space.

Reply:
Indeed, by restoring `initControllerCloudService` bootstrapping on k8s works just fine. See 8b60b19f9248611642a26537e1474b0dbfc6df0e